feature-libs/order/components/order-details/order-consignments.service.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { OrderEntry } from '@spartacus/cart/base/root';
+import { Order, Consignment, ReplenishmentOrder } from '@spartacus/order/root';
+import {
+  completedValues,
+  cancelledValues,
+} from './order-detail-items/order-consigned-entries/order-consigned-entries.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OrderConsignmentsService {
+  consignmentStatus: string[] = [
+    'SHIPPED',
+    'IN_TRANSIT',
+    'DELIVERY_COMPLETED',
+    'DELIVERY_REJECTED',
+    'DELIVERING',
+  ];
```

Comment:
[MEDIUM]
Do we really have to expose this raw array as a public field of the service? If we do it as is, then any consumer of it can potentially:
1. replace this field value with any custom array
2. mutate the array (push new values or remove some)

I believe the array should be made `readonly` and wrapped with `Object.freeze()` so it's not mutable
```suggestion
  readonly consignmentStatuses: ReadonlyArray<string> = Object.freeze([
    'SHIPPED',
    'IN_TRANSIT',
    'DELIVERY_COMPLETED',
    'DELIVERY_REJECTED',
    'DELIVERING',
  ]);
```

Reply:
done